cleanlab/object_detection/summary.py

```diff
@@ -444,6 +450,163 @@ def visualize(
+
+    return counter_dict
+
+
+def get_average_per_class_confusion_matrix(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    num_procs: int = 1,
+    class_names: Optional[List[Union[int, str]]] = None,
+) -> Dict[Union[int, str], Dict[str, float]]:
+    """
+    Compute a confusion matrix dictionary for each class containing the average number of True Positive, False Positive, and False Negative detections from the object detection model across a range of Intersection over Union thresholds.
+
+    The confusion matrices can help you identify model strengths and determine potential biases.
+
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`object_detection.filter.find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`object_detection.filter.find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    num_procs:
+        Number of processes for parallelization. Default is 1.
+
+    class_names:
+        Optional dictionary mapping one-hot-encoded class labels back to their original class names in the format ``{"integer-label": "original-class-name"}``
+
+
+    Returns
+    -------
+    avg_metrics: dict
+        A distionary containing the average confusion matrix.
+
+        The default range of Intersection over Union thresholds is from 0.5 to 0.95 with a step size of 0.05.
+    """
+    iou_thrs = np.linspace(0.5, 0.95, int(np.round((0.95 - 0.5) / 0.05)) + 1, endpoint=True)
+    num_classes = len(predictions[0])
+    if class_names is None:
+        class_names = list(range(num_classes))
+    avg_metrics = {class_num: {"TP": 0.0, "FP": 0.0, "FN": 0.0} for class_num in class_names}
+
+    for iou_threshold in iou_thrs:
+        results_dict = _get_per_class_confusion_matrix_dict_(
+            labels, predictions, iou_threshold, num_procs
+        )
+
+        for class_num in results_dict:
+            tp = results_dict[class_num]["TP"]
+            fp = results_dict[class_num]["FP"]
+            fn = results_dict[class_num]["FN"]
+
+            avg_metrics[class_names[class_num]]["TP"] += tp
+            avg_metrics[class_names[class_num]]["FP"] += fp
+            avg_metrics[class_names[class_num]]["FN"] += fn
+
+    num_thresholds = len(iou_thrs) * len(results_dict)
+    for class_name in avg_metrics:
+        avg_metrics[class_name]["TP"] /= num_thresholds
+        avg_metrics[class_name]["FP"] /= num_thresholds
+        avg_metrics[class_name]["FN"] /= num_thresholds
+    return avg_metrics
+
+
+def calculate_per_class_metrics(
```

Comment:
This is also broken due to issue above. I suggest implementing fix + expending tests to using dictionary of class_names. Otherwise the documentation LGTM after this is fixed, should be good to go.